aodncore/pipeline/steps/harvest.py

```diff
@@ -176,48 +166,69 @@ def execute_talend(self, executor, matched_file_list, talend_base_dir):
         p = SystemProcess(talend_exec, shell=True)
 
         self._logger.info('--- START TALEND OUTPUT ---')
-        # log talend output with basic formatter
         with LoggingContext(self._logger, format_='%(message)s'):
             try:
                 p.execute()
-            except SystemCommandFailedError:
+            except Exception:
                 self._logger.error(p.stdout_text)
-                # Attempt to delete the files that have already been harvested by one or more previous harvesters
-                self.cleanup_on_error()
                 raise
             else:
                 self._logger.info(p.stdout_text)
             finally:
                 self._logger.info('--- END TALEND OUTPUT ---')
 
     def run_deletions(self, harvester_map, tmp_base_dir):
+        """Function to un-harvest and delete files using the appropriate file upload runner
+        Operates in newly created temporary directory as talend requires a non-existant file to perform un-harvesting
+
+
+        :param harvester_map: mapping of harvesters to the PipelineFileCollection to operate on
+        :param tmp_base_dir: temporary directory base for talend operation
+        """
         for harvester, matched_files in harvester_map.items():
             with TemporaryDirectory(prefix='talend_base', dir=tmp_base_dir) as talend_base_dir:
-                matched_file_list = [mf.dest_path for mf in matched_files]
+                self.execute_talend(self._config.trigger_config[harvester]['exec'], matched_files, talend_base_dir)
+
+            files_to_delete = matched_files.filter_by_bool_attribute('pending_store_deletion')
+            if files_to_delete:
+                self.upload_runner.run(files_to_delete)
 
-                self.execute_talend(self._config.trigger_config[harvester]['exec'], matched_file_list, talend_base_dir)
+    def run_undo_deletions(self, harvester_map):
+        """Function to un-harvest and undo stored files as appropriate in the case of errors
+        Operates in newly created temporary directory as talend requires a non-existant file to perform un-harvesting
 
-            for mf in matched_files:
-                mf.is_harvested = True
```

Comment:
This is_harvested attribute has been removed from run_deletions, but should be retained. is_harvested is a bit confusingly named, but means "at least one ran on this file", regardless of whether it was an addition or deletion. Given that fact, this attribute should be set in the generic execute_talend method directly after the subprocess is confirmed to have completed without exception.